signature-specification.md

```diff
@@ -0,0 +1,218 @@
+
+Unless explicitly specified as OPTIONAL, all fields are required. Also, there shouldn’t be any additional fields other than ones specified in JWSPayload, ProtectedHeader, and UnprotectedHeader.
+
+**JWSPayload a.k.a. Claims**:
+Notary v2 is using one private claim(notary) and two public claims(iat and exp). An example of the claim is described below
+```json
+{
+   "notary": {  
+       "subject": {
+           "mediaType": "application/vnd.oci.image.manifest.v1+json",
+           "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+           "size": 16724,
+           "annotations": {
+              "key1": "value1",
+              "key2": "value2",
+               ...
+           }
+       }
+   },
+   "iat": 1234567891000,
+   "exp": 1234567891011
+}
+```
+The payload contains the subject manifest and other attributes that have to be integrity protected.
+
+* **`notary`**(*string-string map*): This is a REQUIRED top-level node and a private claim encapsulating the notary v2 data.
+* **`subject`**(*descriptor*): A REQUIRED manifest that needs to be integrity protected.
+  * **`mediaType`**(*string*): This REQUIRED property contains the media type of the referenced content.
+  * **`size`**(*int64*): This REQUIRED property specifies the size, in bytes, of the raw content.
+  * **`digest`**(*string*): This REQUIRED property is the _digest_ of the targeted content.
+  * **`annotations`**(*string-string map*): This OPTIONAL property contains signed attributes and MUST follow the [annotation rules](https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules). The prefix `org.opencontainers.notary` is reserved for use in Notary v2 and MUST NOT be used outside this specification.
+* **`iat`**(*number*): The REQUIRED property Issued-at(iat) identifies the time at which the signature was issued.
+* **`exp`**(*number*): This OPTIONAL property contains the expiration time on or after which the signature must not be considered valid. 
+
+To leverage JWS claims validation functionality already provided by libraries, we have defined `iat`, `exp` as top-level nodes.
+
+**ProtectedHeaders**:
+Notary v2 supports only three protected headers: alg, cty, and crit.
+```json
+{
+   "alg": "RS256",
+   "cty": "application/vnd.cncf.notary.v2.jws.v1",
+   "crit":["cty"]
+}
+```
+* **`alg`**(*string*): This REQUIRED property defines which algorithm was used to generate the signature. JWS needs an algorithm(alg) to be present in the header, so we have added it as a protected header.
+* **`cty`**(*string*): The REQUIRED property content-type(cty) is used to declare the media type of the secured content(the payload). This will be used to version different variotions of JWS signature. The supported value is `application/vnd.cncf.notary.v2.jws.v0`.
```

Comment:
The version is different than the example.

Reply:
```suggestion
* **`cty`**(*string*): The REQUIRED property content-type(cty) is used to declare the media type of the secured content(the payload). This will be used to version different variations of JWS signature. The supported value is `application/vnd.cncf.notary.v2.jws.v0`.
```